worker/src/main.rs

```diff
@@ -373,7 +383,7 @@ fn worker(
         .expect("Failed to send data to enclave");
 
     // start disk & ipfs snapshotting
-    polkadex_db::start_snapshot_loop();
+    polkadex_db::start_snapshot_loop(api.clone(), eid, genesis_hash);
```

Comment:
I think it's better to only provide the refernce. But leave it as is if it's requring more than one or two changes, this can be solved later.

Reply:
Could you explain what you mean in this comment?

Reply:
pass on &api instead of api.clone()